config/gulp/build.js

```diff
@@ -38,6 +41,51 @@ gulp.task('clean-assets', function (done) {
   );
 });
 
+function spawnP(cmd, args, opts) {
+  return new Promise((resolve, reject) => {
+    console.log(`running ${cmd} ${args}`);
+    child_process.spawn(
+        cmd,
+        args,
+        opts
+      )
+      .on('error', err => {
+        reject(new Error(err));
+      })
+      .on('exit', code => {
+        if (code === 0) {
+          resolve();
+        } else {
+          reject(new Error("Building USWDS failed!"));
+        }
+      });
+  });
+}
+
+// We might be using USWDS from a git repository instead of npm, in
+// which case it might not have the pre-built assets we need. If that's
+// the case, we'll want to build those assets.
+gulp.task('build-uswds-if-needed', function () {
+  const rootDir = path.normalize(path.join(__dirname, '..', '..'));
+  const uswdsDir = path.join(rootDir, 'node_modules', 'uswds');
+  const fractalIndex = path.join(uswdsDir, 'build', 'index.html');
+
+  if (fs.existsSync(fractalIndex)) {
+    dutil.logMessage('build-uswds-if-needed', 'USWDS is already built.');
+    return Promise.resolve();
+  } else {
+    dutil.logMessage('build-uswds-if-needed', 'Building USWDS...');
+
+    //  "federalist": "npm install --dev && gulp build && fractal build",
+    const sharedOpts = { stdio: 'inherit', cwd: uswdsDir };
+    return Promise.all([
+      spawnP('npm', [ 'install', '--only=dev', '--ignore-scripts' ], sharedOpts),
+      spawnP('gulp', ['build', `--cwd ${uswdsDir}` ], sharedOpts),
```

Comment:
Hmm, I think this is effectively running these three commands in *parallel*, but I'm guessing you want to actually run them in serial, based on the comment in line 79?  I think you might want to change this to something like:

```js
return spawnP('npm', [ 'install', '--only=dev', '--ignore-scripts' ], sharedOpts)
  .then(() => spawnP('gulp', ['build', `--cwd ${uswdsDir}` ], sharedOpts))
  .then(() => spawnP('./node_modules/.bin/fractal', [ 'build' ], sharedOpts));
```

More details on the Promise execution timing at https://stackoverflow.com/a/30823708.